trio/_ssl.py

```diff
@@ -331,18 +342,18 @@ class SSLStream(Stream, metaclass=Final):
     # SSLListener.__init__, and maybe the open_ssl_over_tcp_* helpers.
     def __init__(
         self,
-        transport_stream,
-        ssl_context,
+        transport_stream: Stream,
+        ssl_context: _stdlib_ssl.SSLContext,
         *,
-        server_hostname=None,
-        server_side=False,
-        https_compatible=False,
-    ):
-        self.transport_stream = transport_stream
+        server_hostname: str | None = None,
```

Comment:
From `_highlevel_ssl_helpers.open_ssl_over_tcp_stream()`, it seems that `server_hostname` should actually be `str | bytes | None`.

Reply:
Thank you for pointing that out. It turns out there is a bug in typeshed that lead me to thinking its only `str | None`

Type ignore in it's place will be fixed with https://github.com/python/typeshed/pull/10590